### ISSUE TITLE
Version bump loop in workspace

### ISSUE THREAD

**Issue by jbr** [MarcoIeni/release-plz#1181]
## Bug description

I'm trying to use release-plz in [a fairly large workspace](https://github.com/trillium-rs/trillium) and have discovered that every release updates workspace-local dependencies, which in turn makes release-plz think that a patch release is pending. [Here's an example PR](https://github.com/trillium-rs/trillium/pull/511) that was opened immediately after merging a release-plz `chore: release` PR. The primary concern here is that I'd like to avoid issuing patch releases that have zero meaningful changes, especially since there's no way to only release one crate at a time.

- Would you like to work on a fix? [y/n] Possibly? I don't have a sense of what it would entail, but I'm happy to help out with guidance

**Comment by MarcoIeni** [comment]
> every release updates workspace-local dependencies, which in turn makes release-plz think that a patch release is pending.

Yes, you are right. For example, in the PR you linked, trillium-api is updated because trillium went from 0.2.12 to 0.2.13.

> I'd like to avoid issuing patch releases that have zero meaningful changes

If you don't want release-plz to bump that number, you can edit "0.2.12" to "0.2" [here](https://github.com/trillium-rs/trillium/blob/8f90ad0c9cf1e5e5d09379e51e499715826ff24a/api/Cargo.toml#L26). Release-plz will only bump when "0.3.0" is released. Would this solve your issue? If not, do you want a config flag to disable workspace local dependencies update completely?

> I'm using release-plz wrong, or if it's an inappropriate tool for my needs

I want release-plz to support complex workspaces like trillium, so if something is missing, let's work together to implement it!

**Comment by MarcoIeni** [comment]
>  If the most recent version of some crate (eg trillium-api) depends on a feature that was introduced in trillium@0.2.10, it's not necessarily the case that 0.2.* would compile.

You are right 👍 

> I don't at all mind that updating if updating one trillium-* crate makes cargo update other trillium-* crates to the latest patch versions without requiring cargo update.

I didn't get this. Can you explain it again if you think it's important?

> I'd ideally want a way of excluding some commits from marking a workspace crate as in need of release or of easily excluding some crates from a release.

The code that determines if an update is needed is [here](https://github.com/MarcoIeni/release-plz/blob/688f6614ecdf312781bafcd8c81380532381aabe/crates/release_plz_core/src/next_ver.rs#L754). It's one of the most convoluted parts of the codebase, sorry in advance 😅 

Here we can customize this algorithm as we wish. For example, the trillium-api is updated because release-plz compares the Cargo.toml file [here](https://github.com/MarcoIeni/release-plz/blob/688f6614ecdf312781bafcd8c81380532381aabe/crates/release_plz_core/src/package_compare.rs#L23). 

You are saying we should add a configuration value to ignore local packages update in this diff (right now is a simple file comparison, but we could parse the Cargo.toml and check for diffs there).

> If there were a release-required regex ..

So you are proposing to ignore commits that don't respect a certain regex. That's a great feature request, do you mind creating an issue?

> Maybe git-cliff already supports this sort of exclusion?

Git cliff [supports](https://git-cliff.org/docs/configuration/git#filter_commits) it, but we probably need to re-implement it in release-plz, because git-cliff is invoked later.

> Another way of addressing this would be to open one PR per workspace crate, allowing authors to manually ignore crates (close release PRs) that have insufficient changes to merit release.

This is somehow tracked by https://github.com/MarcoIeni/release-plz/issues/1029

> Yet another way would be to allow @release-plz GitHub comments

Yes, we have something similar in mind for https://github.com/MarcoIeni/release-plz/issues/704

**Comment by baszalmstra** [comment]
I think I'm also running into this issue. This PR is created which bumps all crates even though only a single crate was updated (rattler_lock):

https://github.com/mamba-org/rattler/pull/542

Is there an existing workaround that I could use?

**Comment by MarcoIeni** [comment]
uint proposed [here](https://github.com/MarcoIeni/release-plz/issues/1356#issuecomment-2016570542) that release-plz shouldn't update the version if the change is compatible. (E.g. 0.1.0 to 0.1.1).
What do you all think? 🤔
I'm not too sure about this because if 0.1.1 contains a fix, then people should update the version manually which is boring, and people might forget to do it :/

**Comment by jbeaurivage** [comment]
> I'm not too sure about this because if 0.1.1 contains a fix, then people should update the version manually which is boring, and people might forget to do it :/

@MarcoIeni do you mean running `cargo update`?

**Comment by MarcoIeni** [comment]
no, if 0.1.1 contains a fix, your Cargo.toml should be updated, so that you force your downstream dependencies to update their Cargo.lock.

So with "manual update" I was referring to "manually edit Cargo.toml"
